code/mob/living/carbon/human/procs/emote.dm

```diff
@@ -261,12 +261,15 @@
 					G.throw_at(target,5,1)
 					src.visible_message("<b>[src]</B> farts out a...glowstick?")
 
-			if ("salute","bow","hug","wave", "blowkiss")
+			if ("salute","bow","hug","wave", "blowkiss","sidehug")
 				// visible targeted emotes
 				if (!src.restrained())
 					var/M = null
 					if (param)
-						for (var/mob/A in view(null, null))
+						var/range = 8
+						if (act == "hug" || act == "sidehug")
+							range = 1
+						for (var/mob/A in view(range, src))
```

Comment:
I wasn't sure what range to go with here for ranged emotes like "blowkiss", from a performance perspective with view-loops.
I decided to go with 8, but I am open to suggestions.